server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +257,398 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
+
+    final URL url = getLookupsUpdateURL(node);
+
+    LOG.debug("Sending lookups load/drop request to [%s]. Request [%s]", url, lookupsUpdate);
+
+    try (final InputStream result = httpClient.go(
+        new Request(HttpMethod.POST, url)
+            .addHeader(HttpHeaders.Names.ACCEPT, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .addHeader(HttpHeaders.Names.CONTENT_TYPE, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .setContent(smileMapper.writeValueAsBytes(lookupsUpdate)),
+        makeResponseHandler(returnCode, reasonString),
+        lookupCoordinatorManagerConfig.getHostTimeout()
+    ).get()) {
+      if (httpStatusIsSuccess(returnCode.get())) {
+        try {
+          final LookupsStateWithMap response = smileMapper.readValue(result, LookupsStateWithMap.class);
+          LOG.debug(
+              "Update on [%s], Status: %s reason: [%s], Response [%s].", url, returnCode.get(), reasonString.get(),
+              response
+          );
+          return response;
+        } catch (IOException ex) {
+          throw new IOException(
+              String.format("Failed to parse update response from [%s]. response [%s]", url, result),
+              ex
+          );
+        }
+      } else {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+          StreamUtils.copyAndClose(result, baos);
+        }
+        catch (IOException e2) {
+          LOG.warn(e2, "Error reading response");
+        }
+
+        throw new IOException(
+            String.format(
+                "Bad update request to [%s] : [%d] : [%s]  Response: [%s]",
+                url,
+                returnCode.get(),
+                reasonString.get(),
+                StringUtils.fromUtf8(baos.toByteArray())
+            )
+        );
+      }
+    }
+  }
+
+  @VisibleForTesting
+  LookupsStateWithMap getLookupStateForNode(
+      HostAndPort node
+  ) throws IOException, InterruptedException, ExecutionException
+  {
+    final URL url = getLookupsURL(node);
+    final AtomicInteger returnCode = new AtomicInteger(0);
+    final AtomicReference<String> reasonString = new AtomicReference<>(null);
+
+    LOG.debug("Getting lookups from [%s]", url);
+
+    try (final InputStream result = httpClient.go(
+        new Request(HttpMethod.GET, url)
+            .addHeader(HttpHeaders.Names.ACCEPT, SmileMediaTypes.APPLICATION_JACKSON_SMILE),
+        makeResponseHandler(returnCode, reasonString),
+        lookupCoordinatorManagerConfig.getHostTimeout()
+    ).get()) {
+      if (returnCode.get() == 200) {
+        try {
+          final LookupsStateWithMap response = smileMapper.readValue(result, LookupsStateWithMap.class);
+          LOG.debug(
+              "Get on [%s], Status: %s reason: [%s], Response [%s].", url, returnCode.get(), reasonString.get(),
+              response
+          );
+          return response;
+        } catch(IOException ex) {
+          throw new IOException(
```

Comment:
Maybe add `IOE` to some Druid common package, similar to `ISE` and `IAE`? There are a lot of places in Druid where it would be useful, 4 just in this class.